### DISCUSSION TITLE
Is is possible possible to use the default attribute in fn in mojo?

### DISCUSSION THREAD

**Discussion by tic-top** [modular/modular#826] [Q&A] [answered]
Is is possible possible to use the default attribute in fn in mojo?

**Comment by rarebreed** [comment, marked as answer]
Default attribute?  You mean a keyword argument with a default value in a function like this?

```python
fn defaulted_fun(age: Int = 20) -> Int:
    return age * 2
```

or 

```python
def defaulted_def(age = 20):
    return age * 2
```

if that is what you mean, then no.  [Keyword arguments with default values](https://docs.modular.com/mojo/roadmap.html#small-independent-features) are not supported in mojo yet.  

> ## Small independent features
>
> There are a number of features that are missing that are important to round out the language fully, but which don’t depend strongly on other features. These include things like:
> 
> - Keyword arguments in functions: foo(x=42)

This also applies on the calling side.  You can't do this either

```python
fn regular_fn(age: Int) -> Int:
    return age

fn main():
    let doubled = regular_fn(age = 20)  # error here, since keyword args not supported.  uncomment the lines below
    # need to call it without keyword arg
    # let doubled = regular_fn(20)
```

**Reply by tic-top** [reply]
Thank you so much!
